### DISCUSSION TITLE
Angular ArgsTable: No inputs found for this component

### DISCUSSION THREAD

**Discussion by chief10** [storybookjs/storybook#13995] [Help]
**Describe the bug**
Despite following the [instructions for the ArgsTable in MDX files](https://storybook.js.org/docs/angular/writing-docs/doc-blocks), I am consistently seeing the "No inputs found for this component. " message where I would expect the ArgsTable to show up.

**To Reproduce**

1. Use my config
2. Click on element in the side bar and go to the 'Docs' section
3. See error

**Expected behavior**
I expect the ArgsTable to show up

**Screenshots**

**Code snippets**

main.js

```js
module.exports = {
    stories: [
        '../projects/ng-tui/src/lib/**/*.stories.ts',
        '../pages/**/*.stories.ts',
        '../stories/**/*.stories.@(ts|mdx)'
    ],
    addons: [
        '@storybook/addon-actions',
        '@storybook/addon-links',
        '@storybook/addon-notes',
        '@storybook/addon-knobs/register',
        '@storybook/addon-storysource',
        '@storybook/addon-a11y/register',
        '@storybook/addon-docs',
        '@storybook/addon-jest/register'
    ],
};
```

simplified-component

```ts
@Component({
  selector: 'tui-sticky-subnavbar',
  templateUrl: './sticky-subnavbar.component.html',
})
export class TuiStickySubnavbar {
  /**
   * At what point should the component be shown
   */
  @Input() 
  menuVisibleScrollPosition: number = 0;
}
```

Story

```mdx
import { Meta, Story, ArgsTable } from '@storybook/addon-docs/blocks';
import { moduleMetadata } from '@storybook/angular';

import { TuiStickySubnavbarModule } from '@project-root/lib/sticky-subnavbar/sticky-subnavbar.module';
import { TuiStickySubnavbar }from '@project-root/lib/sticky-subnavbar/sticky-subnavbar.component';

<Meta 
  title='Components' 
  decorators={[
    moduleMetadata({
      imports: [
          TuiStickySubnavbarModule
      ],
      declarations: [
          StickySubnavbarExampleComponent
      ]
    })
  ]}
/>


<Story name='Sticky Subnavbar' height='200px'>{{
  template: `
    Not important
  `
  }}
</Story>


## ArgsTable

<ArgsTable of={TuiStickySubnavbar} />
```
package.json
![Screen Shot 2021-01-26 at 4 21 18 PM](https://user-images.githubusercontent.com/2761499/105924164-9915d000-5ff2-11eb-8537-4bc2572cbee1.png)

**System**
 System:
    OS: macOS High Sierra 10.13.6
    CPU: (8) x64 Intel(R) Core(TM) i7-7920HQ CPU @ 3.10GHz
  Binaries:
    Node: 13.7.0 - /usr/local/bin/node
    Yarn: 1.21.1 - /usr/local/bin/yarn
    npm: 6.13.6 - /usr/local/bin/npm
  Browsers:
    Chrome: 88.0.4324.96
    Firefox: 60.9.0
    Safari: 13.1.2
  npmPackages:
    @storybook/addon-a11y: ^6.1.15 => 6.1.15 
    @storybook/addon-actions: ^6.1.15 => 6.1.15 
    @storybook/addon-docs: ^6.1.15 => 6.1.15 
    @storybook/addon-jest: ^6.1.15 => 6.1.15 
    @storybook/addon-knobs: ^6.1.15 => 6.1.15 
    @storybook/addon-links: ^6.1.15 => 6.1.15 
    @storybook/addon-notes: ^5.3.21 => 5.3.21 
    @storybook/addon-storysource: ^6.1.15 => 6.1.15 
    @storybook/addon-viewport: ^6.1.15 => 6.1.15 
    @storybook/addons: ^6.1.15 => 6.1.15 
    @storybook/angular: ^6.1.15 => 6.1.15 
    @storybook/storybook-deployer: ^2.8.7 => 2.8.7 
  npmGlobalPackages:
    @storybook/cli: 6.0.19

**Comment by Yakoust** [comment]
Hi @chief10 ,

I am facing same issue with versions ^6.3.0
Have you found a solution?

**Reply by shilman** [reply]
Here are a few steps to troubleshoot:
- Is compodoc generating `documentation.json`?
- Does it contain docgen information for your component?
- Are you importing it in your `.storybook/preview.js` and registering it?

**Reply by Yakoust** [reply]
Thanks @shilman for your answer. It works great following those steps:

I added @compodoc/compodoc dependency. Then in `.storybook/preview.js` :

```
import { setCompodocJson } from "@storybook/addon-docs/angular";
import docJson from "../documentation.json";
setCompodocJson(docJson);
```

Then, in `package.json` i have those scripts using `Nx`

```
    "docs:json": "compodoc -p ./tsconfig.base.json -e json -d .",
    "storybook": "yarn docs:json && yarn nx run storybook:storybook"
```

**Reply by Yakoust** [reply]
@shilman I have a new issue with this.

In `.storybook/preview.js` I have to load scss files in order to load some stuff such as angular material theme, custom component theme etc.

So I use to do this ([following documentation](https://storybook.js.org/docs/angular/configure/styling-and-css)):

`import '!style-loader!css-loader!sass-loader!./styles.scss';` (also tried [this way](https://storybook.js.org/docs/angular/configure/webpack#extending-storybooks-webpack-config) but still not work)

Scss are loaded and this is working only when `setCompodocJson(docJson);` is removed... Any idea on this?

**Reply by Yakoust** [reply]
Seems ok after upgarding `@compodoc/compodoc` from 1.1.13 to 1.1.14
